docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+Apache Apex, unified stream processing engine enables application developers to process data at very high throughput with low latency. Although the different types of data have different processing needs, the SQL remains a popular and a generic way for processing data. To ensure that existing ETL developers and developers who are well versed with Database applications adopt stream processing application development with ease, integration of SQL with Apex was needed. Being popular Apache project, Apache Calcite was chosen for the integration and to facilitate the use of SQL with Apex.
+
+# Apex-Calcite Integration
+
+Apache Calcite is a highly customizable engine for parsing and planning queries on relational data from various data sources. Apache Calcite provides the functionality of storage independent query optimization and ways to integrate them into other frameworks which would like to take advantage and expose SQL capability to their users. For details about Apache Calcite, please read at [Apache Calcite Website](https://calcite.apache.org/). 
+
+Particularly in SQL on Apex, Calcite processes query and then creates relational algebra to create processing pipelines. These relational algebra processing pipelines are converted to a DAG with a set of operators to perform business logic on streaming data.
+
+![image alt text](images/calcite-apex.png)
+
+Above figure explains how SQL query gets converted to Apex DAG.
+
+1. The received query from a user is parsed by Calcite’s query parser.
+2. The query parser returns the SQL Node Tree to _Query Planner & Optimizer_ which converts the SQL Node Tree to Relation Expression Tree.
+3. This Relation Expression Tree is received by Apache Apex’s SQL module to finally convert to an Apex DAG having series of operators.
+
+One peculiarity that can be seen in above example is that the data source and data destination are not RDBMS systems. This is because Calcite allows Apex to register table sources and destinations as anything which can return a row type results. So a “scan” relational expression gets converted to “KafkaInputOperator + ParseOperator”, a result of which is series of POJOs reflecting a Row Type. Similarly, the “insert” Relational Expression translated to “FormatOperator + FileOutputOperator”.
+
+For Apex-Calcite integration more details, click [here](https://github.com/apache/apex-malhar/tree/master/sql).
```

Comment:
For Apex-Calcite integration more details => For more details about the integration